mmv1/templates/terraform/constants/resource_dns_resource_record_set.go.erb

```diff
@@ -1,19 +1,50 @@
 func rrdatasDnsDiffSuppress(k, old, new string, d *schema.ResourceData) bool {
-	if d.Get("type") == "AAAA" {
-		return ipv6AddressforDnsDiffSuppress(k, old, new, d)
+	oo, nn := d.GetChange("rrdatas")
+	if oo == nil || nn == nil {
+		return false
 	}
-	if d.Get("type") == "MX" {
-		return (strings.ToLower(old) == strings.ToLower(new))
-	}
-	if d.Get("type") == "TXT" {
-		return strings.ToLower(strings.Trim(old, `"`)) == strings.ToLower(strings.Trim(new, `"`))
+
+	olist := convertStringArr(oo.([]interface{}))
+	nlist := convertStringArr(nn.([]interface{}))
+
+	parsefunc := func(x string) string {
+		switch d.Get("type") {
+		case "AAAA":
+			// parse ipv6 to a key from one list
+			return net.ParseIP(x).String()
+		case "MX":
+			return strings.ToLower(x)
+		case "TXT":
+			return strings.ToLower(strings.Trim(x, `"`))
+		default:
+			return x
+		}
 	}
-	return false
+	return rrdatasforDnsDiffSuppress(olist, nlist, parsefunc, d)
 }
 
-func ipv6AddressforDnsDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
-	oldIp := net.ParseIP(old)
-	newIp := net.ParseIP(new)
+type parseStr func(string) string
 
-	return oldIp.Equal(newIp)
+func rrdatasforDnsDiffSuppress(o, n []string, fun parseStr, _ *schema.ResourceData) bool {
+	// compare two lists of unordered records
+	diff := make(map[string]bool, len(o))
+	for _, _o := range o {
```

Comment:
From my understanding `_` is only used in golang for ignored results from functions. We should name  the variable `_o` something more descriptive. Maybe `record` because it represents a single entry in `rrdatas` which is a list of records?